File: src/BaseHmiTypes/Projects/HmiProjectDevice.cs
namespace BaseHmiTypes.Projects;

/// <summary>A device node containing its screens, tags and other device-owned folders.</summary>
public class HmiProjectDevice : HmiProjectFolder
{
    public HmiDeviceInfo Info { get; }

    public HmiProjectDevice(HmiDeviceInfo info, IReadOnlyList<IHmiProjectFolder> folders)
        : base(HmiProjectFolderType.Device, info.Name, info.Id, info.Name,
            getFolders: _ => new ValueTask<IReadOnlyList<IHmiProjectFolder>>(folders))
    {
        Info = info;
    }
}

File: typescript/base-hmi-types/src/projects/HmiProjectDevice.ts
import { HmiDeviceInfo } from "./HmiDeviceInfo.js";
import { HmiProjectFolder } from "./HmiProjectFolder.js";
import { HmiProjectFolderType } from "./HmiProjectFolderType.js";
import { IHmiProjectFolder } from "./IHmiProject.js";

/** A device node containing its screens, tags and other device-owned folders. */
export class HmiProjectDevice extends HmiProjectFolder {
  constructor(readonly info: HmiDeviceInfo, folders: readonly IHmiProjectFolder[]) {
    super(HmiProjectFolderType.Device, info.name, info.id, info.name, folders);
  }
}
